src/internal/schema.ts

```diff
@@ -76,13 +76,13 @@ export function _buildSchema<T, U extends AnyParamConstructor<T>>(
       /** Get Metadata for PreHooks */
       const preHooks: IHooksArray[] = Reflect.getMetadata(DecoratorKeys.HooksPre, cl);
       if (Array.isArray(preHooks)) {
-        preHooks.forEach((obj) => sch.pre(obj.method, obj.func as EmptyVoidFn));
+        preHooks.forEach(obj => sch!.pre(obj.method, obj.func as EmptyVoidFn));
```

Comment:
why is an `!` needed here?

Reply:
Because `sch?` is originally defined as an optional parameter for the function, so it can be assumed to be undefined, but it is definitely assigned above (lines 64-69) so we inform the compiler of that.